custom/ewsghana/reports/email_reports.py

```diff
@@ -0,0 +1,222 @@
+        ])
+
+    @property
+    def rows(self):
+        def percent(x, y):
+            return "%d%% (%d/%d)" % (x * 100 / (y or 1), x, y)
+
+        locations = self.get_locations(self.config['location_id'], self.config['domain'])
+        reported = StockTransaction.objects.filter(
+            case_id__in=locations, report__date__range=[self.config['startdate'],
+                                                        self.config['enddate']]).distinct('case_id').count()
+        completed = 0
+        for loc in locations:
+                products_count = len(SQLLocation.objects.get(supply_point_id=loc).products)
+                st = StockTransaction.objects.filter(
+                    case_id=loc, report__date__range=[self.config['startdate'],
+                                                      self.config['enddate']]).distinct('product_id').count()
+                if products_count == st:
+                    completed += 1
+
+        return [[percent(reported, len(locations)), percent(completed, reported)]]
+
+
+class StockSummaryReportingData(EmailReportingData):
+    def get_locations(self, loc_id, domain):
+        return [loc.supply_point_id for loc in get_stock_summary_locations(loc_id, domain)]
+
+
+class CMSRMSSummaryReportingData(EmailReportingData):
+    def get_locations(self, loc_id, domain):
+        return [loc.supply_point_id for loc in get_rms_cms_summary_locations(domain)]
+
+
+class StockSummaryReport(MultiReport):
+    title = "Weekly Stock Summary Report"
+    fields = [AsyncLocationFilter, DatespanFilter]
+    name = "Stock Summary Report"
+    slug = 'stock_summary_report'
+    exportable = False
+    is_exportable = False
+    split = False
+
+    @property
+    def report_config(self):
+        return dict(
+            domain=self.domain,
+            startdate=self.datespan.startdate_utc,
+            enddate=self.datespan.enddate_utc,
+            location_id=self.request.GET.get('location_id'),
+            program='',
+            products=''
+        )
+
+    @property
+    @memoized
+    def data_providers(self):
+        return [StockSummaryReportingData(self.report_config),
+                StockSummaryReportData(self.report_config)]
+
+    @classmethod
+    def show_in_navigation(cls, domain=None, project=None, user=None):
+        return True
+
+
+class CMSRMSReport(MultiReport):
+    title = "Weekly Stock Summary Report - CMS and RMS"
+    fields = [DatespanFilter]
+    name = "CMS and RMS Summary Report"
+    slug = 'cms_rms_summary_report'
+    exportable = True
+    is_exportable = True
+    split = False
+
+    @property
+    def report_config(self):
+        return dict(
+            domain=self.domain,
+            startdate=self.datespan.startdate_utc,
+            enddate=self.datespan.enddate_utc,
+            location_id=SQLLocation.objects.filter(
+                location_type='country', domain=self.config['domain'])[0].location_id,
```

Comment:
if we find ourselves making this query in more than one place probably makes sense to pull out into a utility function